engine/src/main/scala/cromwell/engine/workflow/tokens/RoundRobinQueueIterator.scala

```diff
@@ -7,8 +7,10 @@ import cromwell.engine.workflow.tokens.TokenQueue.{DequeueResult, LeasedActor}
   * It will keep rotating the list until it finds a queue with an element that can be dequeued.
   * If no queue can be dequeued, the iterator is empty.
   */
-final class RoundRobinQueueIterator(initialTokenQueue: List[TokenQueue], initialPointer: Int)
-    extends Iterator[LeasedActor] {
+final class RoundRobinQueueIterator(initialTokenQueue: List[TokenQueue],
+                                    initialPointer: Int,
+                                    quotaExhaustedGroups: List[String]
```

Comment:
Internally to this class, there's nothing specific about quota. We could suggest future extension with a generic name.
```suggestion
                                    excludedGroups: List[String]
```